src/Algebra/Properties/CancellativeCommutativeSemiring.agda

```diff
@@ -0,0 +1,73 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some theory for CancellativeCommutativeSemiring.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra using (CancellativeCommutativeSemiring)
+import Algebra.GCD
+import Algebra.Properties.Semigroup.Divisibility as SemigroupDiv
+import Algebra.Properties.Semiring as SemiringProp
+open import Data.Product using (_×_; _,_; proj₁; proj₂)
+open import Data.Sum using (_⊎_; inj₁; inj₂)
+open import Function using (case_of_; flip)
+open import Level using (_⊔_)
+open import Relation.Binary using (Rel; Symmetric; Setoid)
+open import Relation.Unary using (Pred)
+
+module Algebra.Properties.CancellativeCommutativeSemiring
+  {α ℓ} (R : CancellativeCommutativeSemiring α ℓ)
+  (open CancellativeCommutativeSemiring R)
+  (0≉1 : 0# ≉ 1#)
```

Comment:
This should not be module parameter as it is not required by all the proofs in this file. Instead it should be an argument to only the proofs that need it.

Reply:
Done.

> Regardless, we should be consistent.
All right, moved to Algebra.Properties.Semiring.Divisibility.